### PR TITLE
MINIFICPP-2386 Add Jom support for building OpenSSL in parallel

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2386

----------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: cmake/BundledOpenSSL.cmake [hunk 1]
```diff
@@ -71,17 +71,37 @@ function(use_openssl SOURCE_DIR BINARY_DIR)
...
+            processorcount(jobs)
+            set(OPENSSL_BUILD_COMMAND ${JOM_EXECUTABLE_PATH} -j${jobs})
+            set(OPENSSL_INSTALL_COMMAND ${JOM_EXECUTABLE_PATH} install)
+            set(OPENSSL_WINDOWS_COMPILE_FLAGS /FS)
+        else()
+            message("Using nmake for OpenSSL build")
+            set(OPENSSL_BUILD_COMMAND nmake)
+            set(OPENSSL_INSTALL_COMMAND nmake install)
+            set(OPENSSL_WINDOWS_COMPILE_FLAGS "")
+        endif()
```
Comment (szaszm) [inline cmake/BundledOpenSSL.cmake:93]: I would just use jom whenever it's installed, and fall back to nmake when it's not.

Comment (lordgamez) [reply cmake/BundledOpenSSL.cmake]: Updated in 36f6133cbd3c5c280d20c3738a7610d45735e11d

**Review by szaszm — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -209,6 +209,13 @@ jobs:
       - name: Add sccache to path
         run: '[Environment]::SetEnvironmentVariable("PATH", [Environment]::GetEnvironmentVariable("PATH", [EnvironmentVariableTarget]::Machine) + ";" + (Split-Path -Path $env:SCCACHE_PATH -Parent), [EnvironmentVariableTarget]::Machine);'
         shell: powershell
+      - name: Install jom
+        run: |
+          Invoke-WebRequest -Uri "http://download.qt.io/official_releases/jom/jom.zip" -OutFile "jom.zip"
+          New-Item -ItemType Directory -Path D:\jom
+          Expand-Archive -Path jom.zip -DestinationPath D:\jom
+          [Environment]::SetEnvironmentVariable("PATH", [Environment]::GetEnvironmentVariable("PATH", [EnvironmentVariableTarget]::Machine) + ";D:\jom", [EnvironmentVariableTarget]::Machine);
+        shell: powershell
```
Comment (adamdebreceni) [inline .github/workflows/ci.yml:218]: should we verify the zip's hash like we do with sqliteodb_w64.exe?

Comment (lordgamez) [reply .github/workflows/ci.yml:219]: Good point, updated in 2ec68cae5d0af75e3dc7219a4164d3d529b6987c

**Review by adamdebreceni — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**